drivers/bluetooth/hci/spi.c

```diff
@@ -302,36 +287,44 @@ static void bt_spi_rx_thread(void)
 	uint8_t size = 0U;
 	int ret;
 	int len;
+	uint8_t success, is_irq_pin_high;
 
 	(void)memset(&txmsg, 0xFF, SPI_MAX_MSG_LEN);
-
+	/* Enable the interrupt line */
+	gpio_pin_interrupt_configure_dt(&irq_gpio, GPIO_INT_EDGE_TO_ACTIVE);
 	while (true) {
-		/* Enable the interrupt line */
-		gpio_init_callback(&gpio_cb, bt_to_active_isr, BIT(irq_gpio.pin));
-		gpio_pin_interrupt_configure_dt(&irq_gpio, GPIO_INT_LEVEL_ACTIVE);
 
 		/* Wait for interrupt pin to be active */
 		k_sem_take(&sem_request, K_FOREVER);
 
-		/* Wait for SPI bus to be available */
-		k_sem_take(&sem_busy, K_FOREVER);
-
 		LOG_DBG("");
 
-		do {
+		/* On BlueNRG-MS, host is expected to read */
+		/* as long as IRQ pin is high */
+		while (irq_pin_high()) {
```

Comment:
`irq_pin_high` is defined as `#define irq_pin_high(...) 0` when `!CONFIG_BT_SPI_BLUENRG`, so this code will now never run with other controllers.

Reply:
I think you are correct, and I will try to find another solution.